pkg/discovery/contextresource_test.go

```diff
@@ -69,8 +76,12 @@ func TestGetContextResourcePathWhenExists(t *testing.T) {
 				defer f.Close()
 			}
 			result, err := GetContextResourcePath(wd)
-			assert.NoError(t, err)
-			assert.Equal(t, filepath.Join(wd, tt.expected), result)
+			if tt.expectErr != nil {
```

Comment:
I'd either create another function to test the error, or assert that the errors are the same without having an if else clause:
`assert.ErrorIs(t, tt.expectedErr, err)`